lib/renderer/src/renderer/scene/rust/vange_rs.h

```diff
@@ -0,0 +1,101 @@
+#ifndef VANGE_RS_H
+#define VANGE_RS_H
+
+#include <cstdint>
+
+/**
+  This is a stub file for the upcoming vange-rs library bindings
+**/
+
+// TODO: agree on how we deal with structs
+struct rv_camera_description {
+	float fov;
+	float aspect;
+	float near;
+	float far;
+};
+
+struct rv_vector3 {
+	float x;
+	float y;
+	float z;
+};
+
+struct rv_quaternion {
+	float x;
+	float y;
+	float z;
+	float w;
+};
+
+struct rv_transform {
+	rv_vector3 position;
+	rv_quaternion rotation;
+};
+
+typedef void* (*rv_gl_functor)(const char*) ;
+
+struct rv_init_descriptor {
+	uint32_t width;
+	uint32_t height;
+	rv_gl_functor gl_functor;
+};
+
+struct rv_map_description {
+	int32_t width; // H_SIZE
+	int32_t height; // V_SIZE
+
+	// data layout: first `width` bytes are height, the rest `width` bytes are meta data
+	uint8_t** lineT;
+
+	// Material offsets in the palette
+	uint8_t* material_begin_offsets;
+	uint8_t* material_end_offsets;
+
+	// 8 for world, 16 for escave
+	int32_t material_count;
+};
+
+// TODO: or it could be left, bottom, right, top
+struct rv_rect {
+	int32_t x;
+	int32_t y;
+	int32_t width;
+	int32_t height;
+};
+
+typedef void* rv_context;
+
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+	extern int32_t rv_api_1;
+
+	rv_context rv_init(rv_init_descriptor desc);

```

Comment:
there is no `rv_resize`?

Reply:
Yes, I didn't implement it yet, but I remember that I should add this one :)

Reply:
@kvark added `rv_api_1` check